File: src/main/java/com/prmncr/normativecontrol/dtos/Document.java
package com.prmncr.normativecontrol.dtos;

import lombok.Getter;
import lombok.RequiredArgsConstructor;
import lombok.Setter;

@RequiredArgsConstructor
public class Document {
    @Getter
    private final String id;
    @Getter
    private final byte[] file;
    @Getter
    @Setter
    private State state = State.QUEUE;
    @Getter
    @Setter
    private Result result;
}

File: src/main/kotlin/com/maeasoftworks/normativecontrol/dtos/Document.kt
package com.maeasoftworks.normativecontrol.dtos

import com.maeasoftworks.normativecontrol.dtos.enums.FailureType
import com.maeasoftworks.normativecontrol.dtos.enums.State

class Document(
    val id: String,
    val accessKey: String,
    var file: ByteArray = ByteArray(0),
    var state: State = State.QUEUE,
    var failureType: FailureType = FailureType.NONE
)